packages/expo-updates/src/Updates.types.ts

```diff
@@ -22,7 +22,37 @@ export enum UpdateEventType {
   ERROR = 'error',
 }
 
-type UpdateCheckResultRollBackToEmbedded = {
+enum UpdateCheckResultNotAvailableReason {
+  /**
+   * No update manifest or rollback directive received from the update server.
+   */
+  UPDATE_NOT_AVAILABLE_ON_SERVER = 'updateNotAvailableOnServer',
+  /**
+   * An update manifest was received from the update server, but the update is not launchable, or does not pass the configured selection policy.
+   */
+  UPDATE_REJECTED_BY_SELECTION_POLICY = 'updateRejectedBySelectionPolicy',
+  /**
+   * An update manifest was received from the update server, but the update has been previously launched on this device and never successfully launched.
+   */
+  UPDATE_PREVIOUSLY_FAILED = 'updatePreviouslyFailed',
+  /**
+   * An update manifest was received from the update server, but the update is not launchable, or does not pass the configured selection policy.
+   */
+  ROLLBACK_REJECTED_BY_SELECTION_POLICY = 'rollbackRejectedBySelectionPolicy',
+  /**
+   * A rollback directive was received from the update server, but this app is not configured to have an embedded update.
+   */
+  ROLLBACK_NO_EMBEDDED_CONFIGURATION = 'rollbackNoEmbeddedConfiguration',
+  /**
+   * A rollback directive was received from the update server, but no embedded manifest was found.
+   */
+  ROLLBACK_NO_EMBEDDED_MANIFEST_FOUND = 'rollbackNoEmbeddedManifestFound',
```

Comment:
IMO we could probably combine these into just a "no embedded manifest" enum case.